api/hypershift/v1beta1/gcp.go

```diff
@@ -81,4 +127,110 @@ type GCPPlatformSpec struct {
+	//
+	// Note: HyperShift generates this value. Validation is intentionally stricter
+	// than Google's minimum (e.g., disallow leading/trailing hyphens) because we
+	// fully control the ID format. If we later allow user-provided IDs, consider
+	// relaxing to ^[a-z0-9-]{4,32}$ and keep the 'gcp-' reserved-prefix check.
+	//
+	// +required
+	// +immutable
+	// +kubebuilder:validation:MinLength=4
+	// +kubebuilder:validation:MaxLength=32
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([a-z0-9-]{2,30}[a-z0-9])$`
+	// +kubebuilder:validation:XValidation:rule="!self.startsWith('gcp-')", message="Pool ID cannot start with reserved prefix 'gcp-'"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Pool ID is immutable"
+	PoolID string `json:"poolID"`
+
+	// providerID is the workload identity provider identifier within the pool.
+	// This provider handles the token exchange between external and GCP identities.
+	// Must be 4-32 characters; allowed: lowercase letters (a-z), digits (0-9), hyphens (-).
+	// Cannot start or end with a hyphen.
+	// The prefix "gcp-" is reserved by Google and cannot be used.
+	//
+	// Note: See poolID above regarding intentionally stricter validation for generated IDs.
+	//
+	// +required
+	// +immutable
+	// +kubebuilder:validation:MinLength=4
+	// +kubebuilder:validation:MaxLength=32
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([a-z0-9-]{2,30}[a-z0-9])$`
+	// +kubebuilder:validation:XValidation:rule="!self.startsWith('gcp-')", message="Provider ID cannot start with reserved prefix 'gcp-'"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Provider ID is immutable"
+	ProviderID string `json:"providerID"`
+
+	// serviceAccountsRef contains references to various Google Service Accounts
+	// required to enable integrations for different controllers and operators.
+	// This follows the AWS pattern of having different roles for different purposes.
+	//
+	// +required
+	// +immutable
+	ServiceAccountsRef GCPServiceAccountsRef `json:"serviceAccountsRef"`
+}
+
+// GCPServiceAccountsRef contains references to Google Service Accounts for different controllers.
+// Each service account should have the appropriate IAM permissions for its specific role.
+type GCPServiceAccountsRef struct {
+	// nodePoolEmail is the Google Service Account email for CAPG controllers
+	// that manage NodePool infrastructure (VMs, networks, disks, etc.).
+	// This GSA needs compute.*, network.*, and storage.* permissions.
+	// Format: service-account-name@project-id.iam.gserviceaccount.com
+	//
+	// +required
+	// +immutable
+	// +kubebuilder:validation:Pattern=`^[a-z][a-z0-9-]{4,28}[a-z0-9]@[a-z][a-z0-9-]{4,28}[a-z0-9]\.iam\.gserviceaccount\.com$`
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=100
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="NodePool email is immutable"
+	NodePoolEmail string `json:"nodePoolEmail"`
```

Comment:
If the identifier for a service account is an email, then maybe just `NodePool` is sufficient here? I think it would make it clearer that this is the service account for NodePools